lib/index.js

```diff
@@ -2108,6 +2108,191 @@ addMethod('toPromise', function (PromiseCtor) {
+        return onNext;
+    }
+
+    // Otherwise create an observer object
+    var observer = {};
+
+    if (_.isFunction(onNext)) {
+        observer.next = onNext;
+    }
+
+    if (_.isFunction(onError)) {
+        observer.error = onError;
+    }
+
+    if (_.isFunction(onComplete)) {
+        observer.complete = onComplete;
+    }
+
+    return observer;
+}
+
+/**
+ * Consumes values using the Observable subscribe signature. Unlike other
+ * consumption methods, subscribe can be called multiple times. Each
+ * subscription will receive the current value before receiving the next value.
+ * Subscribing to an already consumed stream will result in an error.
+ *
+ * Implements the Observable subscribe functionality as defined by the spec:
+ * https://tc39.github.io/proposal-observable/#observable-prototype-subscribe
+ *
+ * @id toPromise
+ * @section Consumption
+ * @name Stream.subscribe(onNext, onError, onComplete)
+ * @param {Function|object|null} onNext - Handler for next value or observer
+ * @param {Function|null} onError - Handler function for errors.
+ * @param {Function|null} onCompleted - Handler Function when stream is done.
+ * @api public
+ *
+ * // with callbacks
+ * _([1, 2, 3, 4]).subscribe(
+ *     function onNext (x) {
+ *         // Called for each value that comes downstream
+ *         console.log('Received onNext value', x);
+ *     },
+ *     function onError (err) {
+ *         // Called one time with error or zero if no errors occur upstream
+ *         console.error('Single highland stream error', err);
+ *     },
+ *     function onComplete () {
+ *         // Receives no arguments
+ *         // Called only once when stream is completed.
+ *         console.log('Completed!');
+ *     }
+ * );
+ *
+ * // with an observer
+ * _([1, 2, 3, 4]).subscribe({
+ *     next (x) {
+ *         console.log('Received next value', x);
+ *     },
+ *     error (err) {
+ *         console.error('An error occurred upstream', err);
+ *     },
+ *     complete () {
+ *         console.log('Completed!')
+ *     }
+ * });
+ */
+
+addMethod('subscribe', function (onNext, onError, onComplete) {
+    var self = this;
+
+    // Don't let users subscribe to an already completed stream
+    if (self.ended) {
+        throw new Error('Subscribe called on an already completed stream.');
+    }
+
+    var observer = createObserver(onNext, onError, onComplete);
+    var subscription = self.fork().consume(function (err, x, push, next) {
+        if (err) {
+            push(null, nil);
+
+            if (_.isFunction(observer.error)) {
+                observer.error(err);
+            }
+        }
+        else if (x === nil) {
+            if (_.isFunction(observer.complete)) {
+                observer.complete();
+            }
+        }
+        else {
+            if (_.isFunction(observer.next)) {
+                observer.next(x);
+            }
+            next();
+        }
+    });
+
+    subscription.resume();
+
+    return {
```

Comment:
The subscription object is missing a `closed` property.

It may be cleaner to define a whole new `ObservableSubscription` class in a different file for this. I've been trying to move away from keeping everything in a single `index.js` file.

Reply:
Ok, in that case should we also create a Subscription class too?

Reply:
`ObservableSubscription` is the same thing as the TC39 `Subscription` class. I just wanted it to be clear that it's meant for Observable compatibility.

Reply:
Thanks for clarifying.